boot/freeldr/bootsect/ext.S

```diff
@@ -0,0 +1,490 @@
+    xor bx, bx                                          // Clear BX
+    ret
+
+/* Read the Inode in EAX register */
+ExtReadInode:
+    xor edx, edx										// Clear EDX before division
+    dec eax												// Inode = Inode -1
+    div dword ptr BP_REL(ExtInodesPerGroup)				// Inode /= ExtInodesPerGroup
+    mov dword ptr BP_REL(ExtReadInodeGroup), eax		// Store the Inode group
+    mov dword ptr BP_REL(ExtReadInodeIndex), edx		// Store the Inode index
+
+    xor edx, edx 										// Clear EDX before multiplication
+    mul dword ptr BP_REL(ExtGroupDescSize)				// Inode group *= ExtGroupDescSize
+    mov dword ptr BP_REL(ExtReadInodeGroup), eax		// Store the precalculated Inode group
+
+    xor edx, edx										// Clear EDX before multiplication
+    mov eax, dword ptr BP_REL(ExtReadInodeIndex)		// Get the read Inode index
+    mul dword ptr BP_REL(ExtInodeSize)					// Inode group *= ExtInodeSize
+    mov dword ptr BP_REL(ExtReadInodeIndex), eax		// Store the Inode index
+
+    /* Calculate the Inode index block */
+    mov si, offset ExtReadInodeIndexBlock - start
+    call ExtCalculateBlock
+
+    /* Calculate the Inode group block */
+    mov eax, dword ptr BP_REL(ExtReadInodeGroup)
+    mov si, offset ExtReadInodeGroupBlock - start
+    call ExtCalculateBlock
+
+    /* Calculate the Inode group offset */
+    mov bx, offset ExtReadInodeGroup - start
+    mov si, offset ExtReadInodeGroupBlock - start
+    mov di, offset ExtReadInodeGroupOffset - start
+    call ExtCalculateOffset
+
+    /* Calculate the Inode index offset */
+    mov bx, offset ExtReadInodeIndex - start
+    mov si, offset ExtReadInodeIndexBlock - start
+    mov di, offset ExtReadInodeIndexOffset - start
+    call ExtCalculateOffset
+
+    /* Read group descriptor */
+    call ExtReadGroupDescriptor
+
+    /* Set the offset address */
+    mov si, word ptr BP_REL(ExtReadInodeGroupOffset)
+
+    /* Get InodeTable field from the ExtGroupDescriptor structure */
+    mov eax, dword ptr es:[si + 8]
+
+    /* Sum EAX with Inode index block */
+    add eax, dword ptr BP_REL(ExtReadInodeIndexBlock)
+
+    jmp ExtReadBlock
+
+msgDiskError:
+    .ascii "Disk error", CR, LF, NUL
+msgAnyKey:
+    .ascii "Press key", CR, LF, NUL
+
+    .org 509
+
+BootPartition:
+    .byte 0
+
+    .word HEX(0aa55)       // BootSector signature
+
+// End of bootsector
+//
+// Now starts the extra boot code that we will store
+// at sector 1 on a EXT2 volume
+
+LoadRootDirectory:
+    xor eax, eax
+    mov al, EXT_ROOT_INO			// Put the root directory inode number in EAX
+    call ExtReadInode				// Read the inode
+
+    call ReadDirectPointers         // Read the direct pointers
+    call FindFile                   // Find the extended loader and run it
+
+    jmp ExtLdrPrintFileNotFound
+
+ReadDirectPointers:
+    push es
+    pushad
+    xor eax, eax                                        // Clear eax register because it will be used for multiplication
+    call ExtSetInodeSegment
```

Comment:
How many bytes of opcodes do you get by doing this call, instead of "inlining" what the function does?
(i.e.:
```asm
mov bx, EXT_INODE_ADDRESS / 16
mov es, bx                                          // Set ES
// xor bx, bx // May not be needed.
```

I also have the same question for all these other ExtSet...Segment functions (_excepting the `ExtSetFileSegment` helper_) and calls to them used elsewhere in the code.